libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -1362,106 +1365,571 @@ def _extract_images_from_page(self, page: pypdfium2._helpers.page.PdfPage) -> st
+        page: pdfplumber.page.Page,
+    ) -> list[list[list[Any]]]:
+        """Extract tables from a PDF page.
+
+        Args:
+            page: The PDF page to extract tables from.
 
-        return extract_from_images_with_rapidocr(images)
+        Returns:
+            A list of tables, where each table is a list of rows, and each row is a
+            list of cell values.
+        """
+        if not self.extract_tables:
+            return []
+        table_settings = self.extract_tables_settings
+        tables_list = page.extract_tables(table_settings)
+        return tables_list
+
+    def _convert_table(self, table: list[list[str]]) -> str:
+        """Convert a table to the specified format.
+
+        Args:
+            table: The table to convert.
+
+        Returns:
+            The table content as a string in the specified format.
+        """
+        format = self.extract_tables
+        if format is None:
+            return ""
+        if format == "markdown":
+            return self._convert_table_to_markdown(table)
+        elif format == "html":
+            return self._convert_table_to_html(table)
+        elif format == "csv":
+            return self._convert_table_to_csv(table)
+        else:
+            raise ValueError(f"Unknown table format: {format}")
+
+    def _convert_table_to_csv(self, table: list[list[str]]) -> str:
+        """Convert a table to CSV format.
+
+        Args:
+            table: The table to convert.
+
+        Returns:
+            The table content as a string in CSV format.
+        """
+        if not table:
+            return ""
+
+        output = ["\n\n"]
+
+        # skip first row in details if header is part of the table
+        # j = 0 if self.header.external else 1
+
+        # iterate over detail rows
+        for row in table:
+            line = ""
+            for i, cell in enumerate(row):
+                # output None cells with empty string
+                cell = "" if cell is None else cell.replace("\n", " ")
+                line += cell + ","
+            output.append(line)
+        return "\n".join(output) + "\n\n"
+
+    def _convert_table_to_html(self, table: list[list[str]]) -> str:
+        """
+        Convert table content as a string in HTML format.
+        If clean is true, markdown syntax is removed from cell content.
+
+        Args:
+            table: The table to convert.
+
+        Returns:
+            The table content as a string in HTML format.
+        """
+        if not len(table):
+            return ""
+        output = "<table>\n"
+        clean = True
+
+        # iterate over detail rows
+        for row in table:
+            line = "<tr>"
+            for i, cell in enumerate(row):
+                # output None cells with empty string
+                cell = "" if cell is None else cell.replace("\n", " ")
+                if clean:  # remove sensitive syntax
+                    cell = html.escape(cell.replace("-", "&#45;"))
```

Comment:
Add a doc-string about the presence of replacement for `-`. I don't think most developers will know the character set that's covered by html.escape. Is escaping the `-` necessary?